src/time/location/loader.cr

```diff
@@ -41,6 +48,23 @@ class Time::Location
     end
   end
 
+  # :nodoc:
+  def self.load_from_android_tzdata(name : String, path : String) : Time::Location?
+    return nil unless File.exists?(path)
+
+    mtime = File.info(path).modification_time
+    if (cache = @@location_cache[name]?) && cache[:time] == mtime
+      cache[:location]
+    else
+      File.open(path) do |file|
+        read_android_tzdata(file, false) do |location_name, location|
+          @@location_cache[location_name] = {time: mtime, location: location}
```

Comment:
Why is the modification time cached per location entry? They're all loaded from the same file, so shouldn't it be a global property?

Reply:
IMO it is easier to reuse the same cache mechanism as the individual zoneinfo files

Reply:
Okay. Let's do it that way.